### ISSUE TITLE
Rule S4275: false positive on Visual Basic when using property overload

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#5336]
### Description

S4275 raises a false positive when analyzing overridden properties in Visual Basic code.

### Repro steps

   Public Class A
        Private _chargeId as Integer

        Protected Overridable Property ChargeId As Integer
            Get
                Return _chargeId
            End Get
            Set(value As Integer)
                If (value <> _chargeId)
                    _chargeId = value ' strictly speaking, we're doing other validation, but a scenario where auto properties aren't a solution
                End If
            End Set
        End Property
    End Class

    Public Class B
        Inherits A

        Protected Overrides Property ChargeId As Integer
            Get ' Noncompliant FP
                Return If(true, MyBase.ChargeId, 1234)
            End Get
            Set(value As Integer) ' Noncompliant FP
                MyBase.ChargeId = value
            End Set
        End Property
    End Class

### Expected behavior

We should not raise issues in this case since `_chargeId` is not accessible in the derived class.

### Actual behavior

False positives are raised.

### Related information

* C#/VB.NET Plugins version: [8.34](https://github.com/SonarSource/sonar-dotnet/releases/tag/8.34.0.42011)

**Comment by eXpl0it3r** [comment]
Just ran into this on a C# code base using "Community Editionv9.9.8 (build 100196)"

```csharp
public class MyBase : ViewModelBase
{
    private int _unit;
    public virtual int Unit
    {
        get => _unit;
        protected set => Set(ref _unit, value);
    }
}

public class Derived : MyBase
{
    public int Unit
    {
        get => base.Unit;
        protected set => base.Unit = value;
    }
}
```

> Refactor this getter so that it actually refers to the field '_unit'.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1867](https://sonarsource.atlassian.net/browse/NET-1867)

[NET-1867]: https://sonarsource.atlassian.net/browse/NET-1867?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
